include/pagmo/algorithms/mbh.hpp

```diff
@@ -0,0 +1,374 @@
+     * @param[in] seed seed used by the internal random number generator (default is random)
+     * @throws std::invalid_argument if \p stop is not in (0,1]
+     */
+    template <typename T>
+    explicit mbh(T &&a, unsigned int stop, double perturb, unsigned int seed = pagmo::random_device::next())
+        : algorithm(std::forward<T>(a)), m_stop(stop), m_perturb(1, perturb), m_e(seed), m_seed(seed), m_verbosity(0u),
+          m_log()
+    {
+        if (perturb > 1. || perturb <= 0.) {
+            pagmo_throw(std::invalid_argument, "The scalar perturbation must be in (0, 1], while a value of "
+                                                   + std::to_string(perturb) + " was detected.");
+        }
+    }
+    /// Constructor
+    /**
+     * Constructs Monotonic Basin Hopping using a vector perturbation
+     *
+     * @param[in] a Any object that can construct a pagmo::algorithm and that will
+     * @param[in] stop consecutive runs of the inner algorithm that need to
+     * result in no improvement for pagmo::mbh to stop
+     * @param[in] perturb a vector_double with the perturbations to be applied to each component
+     * of the decision vector of the best population found when generating a new starting point.
+     * These are defined relative to the corresponding bounds.
+     * @param[in] seed seed used by the internal random number generator (default is random)
+     * @throws std::invalid_argument if \p stop is not in (0,1]
+     */
+    template <typename T>
+    explicit mbh(T &&a, unsigned int stop, vector_double perturb, unsigned int seed = pagmo::random_device::next())
+        : algorithm(std::forward<T>(a)), m_stop(stop), m_perturb(perturb), m_e(seed), m_seed(seed), m_verbosity(0u),
+          m_log()
+    {
+        if (!std::all_of(perturb.begin(), perturb.end(), [](double item) { return (item > 0. && item <= 1.); })) {
+            pagmo_throw(std::invalid_argument,
+                        "The perturbation must have all components in (0, 1], while that is not the case.");
+        }
+    }
+    /// Algorithm evolve method (juice implementation of the algorithm)
+    /**
+     * Evolves the population up to when \p stop consecutve runs of the internal
+     * algorithm do not improve the solution.
+     *
+     * @param[in] pop population to be evolved
+     * @return evolved population
+     * @throws std::invalid_argument if the problem is multi-objective or stochastic
+     * @throws std::invalid_argument if the perturbation vector size does not equal the problem size
+     */
+    population evolve(population pop) const
+    {
+        // We store some useful variables
+        const auto &prob = pop.get_problem(); // This is a const reference, so using set_seed for example will not be
+                                              // allowed (pop.set_problem_seed is)
+        auto dim = prob.get_nx();             // This getter does not return a const reference but a copy
```

Comment:
:) I guess it was there to remind me why above its const auto & and there is only auto ....